### PR TITLE
stage2 AArch64: more support for PIE targets (Mach-O)

### PR DESCRIPTION
**Pull request by joachimschmidt557**
This enables a handful of behavior tests previously disabled for `aarch64-macos`.

cc @kubkon

### CODE REVIEW

File: src/arch/aarch64/Emit.zig [hunk 1]
```diff
@@ -209,17 +210,9 @@ fn instructionSize(emit: *Emit, inst: Mir.Inst.Index) usize {
     switch (tag) {
         .load_memory_got,
         .load_memory_direct,
+        .load_memory_ptr_got,
+        .load_memory_ptr_direct,
```
Comment (kubkon) [inline src/arch/aarch64/Emit.zig:214]: Why do we need all 4? Would just `load_memory_got` and `load_memory_direct` not suffice? Or do you want to flag a potential for having a sequence `adrp, ldr` vs `adrp, add, ldr`?

Comment (joachimschmidt557) [reply src/arch/aarch64/Emit.zig:214]: For types that don't fit into a register, the `adrp, ldr` sequence doesn't work, right? In those cases we emit `adrp, add` and `memcpy`.

Comment (kubkon) [reply src/arch/aarch64/Emit.zig:214]: Mhm, right. Also, I've seen multiple times cases of `adrp, add, ldr`, or `adrp, ldr, ldr`. The later sometimes qualifies for an instruction rewrite on the linker side depending on the relocation type of the pair `adrp, ldr` preceding `ldr`.

Comment (joachimschmidt557) [reply src/arch/aarch64/Emit.zig:214]: I'm unsure about the case `adrp, add, ldr`, but the case `adrp, ldr, ldr` seems to me like the contents of the memory location itself is a pointer to something else (maybe a global variable `x` which is a `*u64` and we want to access `x.*` ?)

Comment (kubkon) [reply src/arch/aarch64/Emit.zig:214]: Yep, correct. However, MachO does certain optimisations such as using a different mechanism for TLS pointer loading, which implies a rewrite of `adrp, ldr, ldr` into `adrp, add, ldr` (note that the linker doesn't touch a non-relocated intstruction, in this case the last `ldr`). Anyhow, I digress :-)

Comment (joachimschmidt557) [reply src/arch/aarch64/Emit.zig:214]: At the moment, the only time `load_memory_ptr_got` and `load_memory_ptr_direct` are emitted are for memcpy. If there is a `adrp, add, ldr` sequence I'm pretty sure that the last `ldr` is unrelated to loading the address.

Comment (kubkon) [reply src/arch/aarch64/Emit.zig:214]: Fair, but something to bear in mind for sure though.

**Pull request merged**